plugin/batch-queue/rm_pmi.cpp

```diff
@@ -138,6 +138,17 @@ extern "C" int PMI_Init( int *spawned )
     return ret;
 }
 
+// Intel MPI uses dlopen() and dlsym() to find pmi functions.
+// We need to redirect PMI_Init() to our wrapper, so that
+// pmi is correctly initialized.
+extern "C" void *dlsym(void *handle, const char *symbol)
+{
+  if (strcmp(symbol, "PMI_Init") == 0) {
```

Comment:
Let's change this to:
```C
if (symbol && strncmp(...) == 0) {
  ...
}
```
Do we need to add cases for other `PMI_*` functions? If yes, then how about just returning a handle to the batchqueue plugin in the `dlopen()` call?